whylog/tests/tests_config/test_get_search_range.py

```diff
@@ -332,13 +319,51 @@ def test_search_range_mixed_constraint_type(self):
                 }
             },
             {
+                'clues_groups': [[2, 3], [0, 2]],
+                'name': 'identical',
+                'params': {}
+            },
+        ]
+        rule = Rule([self.cause2, self.cause2], self.effect, constraints1, Rule.LINKAGE_AND)
+        calculated_ranges = self.calculate_range([rule])
+
+        expected_ranges = {
+            'apache': {
+                'date': {
+                    'left_bound': self.earliest_date,
+                    'right_bound': self.effect_time
```

Comment:
Nope, in OR linkage when we have a non primary key group in constraint we must parse file from begin to end